pkg/cloudprovider/util.go

```diff
@@ -0,0 +1,78 @@
+/*
```

Comment:
This package seems like the wrong home. WDYT about `scheduling`?

Reply:
There's an import cycle again if it's moved there.  The problem is that the NodeIsReady now takes an instance type to know about possible extended resources, which is a cloudprovider type.

Reply:
Ok. If the method is deeply tied to an instance type, cloud provider sounds like a good home. I'm a bit wary of util sprawl, so something to keep in mind.